sound/soc/soc-topology.c

```diff
@@ -1567,7 +1544,6 @@ static int soc_tplg_dapm_widget_create(struct soc_tplg *tplg,
 	}
 
 	widget->dobj.type = SND_SOC_DOBJ_WIDGET;
-	widget->dobj.widget.kcontrol_type = kcontrol_type;
```

Comment:
so now this field will remain uninitialised, i.e. set to 0 by `memset(&template, 0,...)` and 0 is an invalid kcontrol type, is that ok?

Reply:
this is related to that previously there was assumption, that only 1 type off control type is associated with the widget... so what this whole PR is trying to change :) So it should not matter at all what the value is. I tried to search for it and I think the only place it was ever used was in sof (and I'm changing it here).